rules/auth0_rules/auth0_custom_role_created.py

```diff
@@ -0,0 +1,53 @@
+from global_filter_auth0 import filter_include_event
+from panther_auth0_helpers import auth0_alert_context, is_auth0_config_event
+from panther_base_helpers import deep_get
+
+
+def rule(event):
+    if not filter_include_event(event):
+        return False
+    data_description = deep_get(event, "data", "description", default="<NO_DATA_DESCRIPTION_FOUND>")
+    return all(
+        [
+            data_description == "Create a role",
+            is_auth0_config_event(event),
+        ]
+    )
+
+
+def title(event):
+    user = deep_get(
+        event, "data", "details", "request", "auth", "user", "email", default="<NO_USER_FOUND>"
+    )
+    request_body_name = deep_get(
+        event, "data", "details", "request", "body", "name", default="<NO_REQUEST_NAME_FOUND>"
+    )
+    request_body_description = deep_get(event, "data", "details", "request", "body", default=[-1])
+
+    if "admin" in request_body_description or "admin" in request_body_name:
+        role_type = "admin"
+    else:
+        role_type = "custom"
+
+    p_source_label = deep_get(event, "p_source_label", default="<NO_P_SOURCE_LABEL_FOUND>")
+    return (
+        f"Auth0 User [{user}] created a "
+        f"role [{request_body_name}] with [{role_type}] "
+        f"permissions in your tenant [{p_source_label}]."
+    )
+
+
+def severity(event):
+    request_body_name = deep_get(
+        event, "data", "details", "request", "body", "name", default="<NO_REQUEST_NAME_FOUND>"
+    )
+    request_body_description = deep_get(
```

Comment:
Here's how I'm reading this logic and the test cases: 

* It seems to me that `request_body_name` is never getting evaluated by any of the test cases. 
* I also think that `request_body_description`'s `default=` kwarg should be a string, and not a list. 

I would recommend either removing `request_body_description` or creating test cases that exercise the logic relating to request_body_name on line 47. 

Here are the values of `.data.details.request.body` in the current test cases 

```shell
user@computer:~/Sources/PAN/panther-analysis $ yq  '.Tests[] | {.Name: .Log.data.details.request.body}' rules/auth0_rules/auth0_custom_role_created.yml
Other Event:
  AfterAuthentication: false
Custom Role Created:
  name: test
  description: test
Admin Role Created:
  name: admin
  description: admin
```

Reply:
> It seems to me that request_body_name is never getting evaluated by any of the test cases.

Added a test for validating the name logic is triggered!

> I also think that request_body_description's default= kwarg should be a string, and not a list.

Correct, updated!

> I would recommend either removing request_body_description or creating test cases that exercise the logic relating to request_body_name on line 47.

Hm are you talking about testing the generated severity on line 47?